sklearn/metrics/_ranking.py

```diff
@@ -379,10 +380,14 @@ def det_curve(y_true, y_score, pos_label=None, sample_weight=None):
 def _binary_roc_auc_score(y_true, y_score, sample_weight=None, max_fpr=None):
     """Binary roc auc score."""
     if len(np.unique(y_true)) != 2:
-        raise ValueError(
-            "Only one class present in y_true. ROC AUC score "
-            "is not defined in that case."
+        warnings.warn(
+            (
+                "Only one class present in y_true. ROC AUC score "
+                "is not defined in that case."
```

Comment:
```suggestion
                "is not defined in that case. The score is set to "
                "0.0."
```

Reply:
I'm wondering if this is a case that we should also have a parameter to control the returned value as with `zero_division`.